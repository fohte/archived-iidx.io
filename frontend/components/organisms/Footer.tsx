import { Icon } from 'antd'

const originalCopyrightYear = 2018

const getCopyrightYears = (): string => {
  const currentYear = new Date().getFullYear()

  if (currentYear === originalCopyrightYear) {
    return `${originalCopyrightYear}`
  }

  return `${originalCopyrightYear}-${currentYear}`
}

const Footer: React.SFC = () => (
  <>
    <Icon type="copyright" /> {getCopyrightYears()} iidx.io
  </>
)

export default Footer
