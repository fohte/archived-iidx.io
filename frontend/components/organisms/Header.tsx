import { Menu } from 'antd'
import Link from 'next/link'
import styled from 'styled-components'

import Brand from '@app/components/atoms/Brand'
import UserNav from '@app/components/molecules/UserNav'
import withAuthState from '@app/lib/withAuthState'

const Header: React.SFC = () => (
  <Menu theme="dark" style={{ lineHeight: '64px' }}>
    <Link href="/" prefetch>
      <BrandWrapper>
        <Brand />
      </BrandWrapper>
    </Link>
    <UserNavWrapper>
      <EnhancedUserNav />
    </UserNavWrapper>
  </Menu>
)

const BrandWrapper = styled.div`
  float: left;
`

const UserNavWrapper = styled.div`
  float: right;
`

const EnhancedUserNav = withAuthState()(UserNav)

export default Header
