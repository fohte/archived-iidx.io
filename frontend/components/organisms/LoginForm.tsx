import { Button, Form, Icon, Input } from 'antd'
import { FormComponentProps } from 'antd/lib/form'
import * as React from 'react'

import InputIcon from '@app/components/atoms/InputIcon'
import withSubmitHandling, {
  ComponentExternalProps,
  InjectedProps,
} from '@app/lib/withSubmitHandling'

export type ExternalProps = ComponentExternalProps<FormValues>

export interface Props
  extends ExternalProps,
    FormComponentProps,
    InjectedProps {}

export interface State {
  submitting: boolean
  errorMessage: string | null
}

export interface FormValues {
  email: string
  password: string
}

const LoginForm: React.SFC<Props> = ({
  form: { getFieldDecorator },
  errorMessage,
  handleSubmit,
  submitting,
}) => (
  <Form onSubmit={handleSubmit}>
    <Form.Item>
      {getFieldDecorator('email', {
        rules: [{ required: true, message: 'Please input your email.' }],
      })(<Input prefix={<InputIcon type="mail" />} placeholder="Email" />)}
    </Form.Item>
    <Form.Item>
      {getFieldDecorator('password', {
        rules: [{ required: true, message: 'Please input your password.' }],
      })(
        <Input
          prefix={<InputIcon type="lock" />}
          type="password"
          placeholder="Password"
        />,
      )}
    </Form.Item>
    <Form.Item>
      <Button type="primary" htmlType="submit">
        {submitting ? <Icon type="loading" /> : 'Log in'}
      </Button>
      {errorMessage}
    </Form.Item>
  </Form>
)

const WrappedNormalLoginForm = Form.create<ExternalProps>()(
  withSubmitHandling<FormValues>()(LoginForm),
)

export default WrappedNormalLoginForm
