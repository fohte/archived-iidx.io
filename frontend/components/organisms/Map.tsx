import * as React from 'react'
import { Segment } from 'semantic-ui-react'

import MapTitleHeader from '@app/components/molecules/MapTitleHeader'
import ResultTable, { Result } from '@app/components/molecules/ResultTable'
import { FindMapMap, FindMapMusic } from '@app/queries'

export type Props = {
  music: FindMapMusic
  map: FindMapMap
  result?: Result
}

const Map: React.SFC<Props> = ({ music, map, result }) => (
  <>
    <Segment basic textAlign="center">
      <MapTitleHeader music={music} map={map} />
    </Segment>

    <Segment basic>
      <ResultTable map={map} result={result} />
    </Segment>
  </>
)

export default Map
