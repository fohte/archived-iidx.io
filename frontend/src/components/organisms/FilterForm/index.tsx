import { faTimes } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as classnames from 'classnames/bind'
import * as _ from 'lodash'
import * as React from 'react'
import { Field as FinalField, Form as FinalForm } from 'react-final-form'
import { toast } from 'react-toastify'

import Button from '@app/components/atoms/Button'
import Checkbox from '@app/components/atoms/Checkbox'
import Container from '@app/components/atoms/Container'
import FormGroup from '@app/components/atoms/FormGroup'
import { Difficulty } from '@app/queries'

import * as css from './style.scss'

const cx = classnames.bind(css)

export type FormValues = {
  title?: string | null
  difficulties: Difficulty[]
  levels: number[]
}

export type Props = {
  initialValues: FormValues
  onSubmit: (values: FormValues) => void
  onCloseRequested: () => void
}

const FilterForm: React.SFC<Props> = ({
  initialValues,
  onCloseRequested,
  onSubmit,
}) => {
  // このコンポーネントが開いている間は背景のスクロールを無効化する
  React.useEffect(() => {
    const { top, position } = document.body.style

    // `documentElement` は古いブラウザで未サポートなので `body` に fallback する
    // @see https://ginpen.com/2017/10/06/chrome-61-scrolltop/
    const scrollTop =
      document.documentElement.scrollTop || document.body.scrollTop

    // position を変更すると背景が最上部までスクロールされるので
    // 擬似的に背景位置を固定する
    document.body.style.top = `-${scrollTop}px`

    document.body.style.position = 'fixed'

    return () => {
      document.body.style.top = top
      document.body.style.position = position

      // position を変更すると背景が最上部までスクロールされるので
      // スクロール位置をコンポーネント表示前のスクロール位置に戻す
      document.documentElement.scrollTop = document.body.scrollTop = scrollTop
    }
  })

  return (
    <FinalForm onSubmit={onSubmit} initialValues={initialValues}>
      {({ form, handleSubmit, hasSubmitErrors, submitError }) => {
        if (hasSubmitErrors) {
          toast.error(submitError)
        }

        return (
          <form
            className={cx('result-search-form')}
            onSubmit={e => {
              handleSubmit(e)
              onCloseRequested()
            }}
          >
            <div className={cx('modal', 'filter-form')}>
              <header className={cx('header')}>
                <Container className={cx('container')}>
                  <h1>Filters</h1>
                  <div
                    className={cx('close-button')}
                    onClick={() => {
                      onCloseRequested()
                    }}
                  >
                    <FontAwesomeIcon icon={faTimes} />
                  </div>
                </Container>
              </header>

              <Container className={cx('container')}>
                <FormGroup
                  label="Level"
                  labelClassName={cx('form-group-label')}
                >
                  <div className={cx('level-field-group')}>
                    {_.rangeRight(1, 13).map(level => (
                      <FinalField
                        key={level}
                        type="checkbox"
                        name="levels"
                        value={level}
                      >
                        {({ input }) => (
                          <div className={cx('form-field')}>
                            <Checkbox {...input}>☆{level}</Checkbox>
                          </div>
                        )}
                      </FinalField>
                    ))}
                  </div>
                </FormGroup>

                <FormGroup
                  label="Difficulty"
                  labelClassName={cx('form-group-label')}
                >
                  {_.map(Difficulty, difficulty => (
                    <FinalField
                      key={difficulty}
                      type="checkbox"
                      name="difficulties"
                      value={difficulty}
                    >
                      {({ input }) => (
                        <div className={cx('form-field')}>
                          <Checkbox {...input}>{difficulty}</Checkbox>
                        </div>
                      )}
                    </FinalField>
                  ))}
                </FormGroup>
              </Container>

              <footer className={cx('footer')}>
                <Container className={cx('container')}>
                  <div className={cx('footer-button-group')}>
                    <Button
                      className={cx('clear-button')}
                      expand={false}
                      type="button"
                      onClick={() => {
                        form.reset({
                          difficulties: [],
                          levels: [],
                        })
                      }}
                    >
                      Clear
                    </Button>
                    <Button color="primary" type="submit">
                      Submit
                    </Button>
                  </div>
                </Container>
              </footer>
            </div>
          </form>
        )
      }}
    </FinalForm>
  )
}

export default FilterForm
