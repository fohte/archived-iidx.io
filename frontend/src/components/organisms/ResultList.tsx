import * as _ from 'lodash'
import ErrorPage from 'next/error'
import * as React from 'react'
import { Divider } from 'semantic-ui-react'

import ResultTable from '@app/components/molecules/ResultTable'
import ResultSearchForm, {
  FormValues,
} from '@app/components/organisms/ResultSearchForm'
import { GetUserResultsComponent } from '@app/queries'
import { Router } from '@app/routes'

export type Props = {
  screenName: string
  initialValues: FormValues
}

const ResultList: React.SFC<Props> = ({ screenName, initialValues }) => {
  const [formValues, setFormValues] = React.useState<FormValues>(initialValues)

  return (
    <>
      <ResultSearchForm
        initialValues={initialValues}
        onSubmit={values => {
          setFormValues(values)
        }}
      />
      <Divider />
      <GetUserResultsComponent
        variables={{
          username: screenName,
          title: formValues.title,
          playStyle: formValues.playStyle,
          difficulty: formValues.difficulties[0],
          levels: formValues.levels,
        }}
      >
        {({ loading, error, data }) => {
          if (loading) {
            return 'loading'
          }
          if (error || !data || !data.searchMaps) {
            return <ErrorPage statusCode={404} />
          }

          return (
            <ResultTable
              showMapData
              maps={_.map(
                data.searchMaps,
                ({ bestResult: result, ...maps }) => ({ ...maps, result }),
              )}
              onClickRow={({ music, playStyle, difficulty }) => {
                if (screenName && music) {
                  Router.pushRoute('map', {
                    screenName,
                    musicId: music.id,
                    playStyle: playStyle.toLowerCase(),
                    difficulty: difficulty.toLowerCase(),
                  })
                }
              }}
            />
          )
        }}
      </GetUserResultsComponent>
    </>
  )
}

export default ResultList
