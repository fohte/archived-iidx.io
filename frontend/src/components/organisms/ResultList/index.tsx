import * as classnames from 'classnames/bind'
import * as _ from 'lodash'
import ErrorPage from 'next/error'
import * as React from 'react'

import Container from '@app/components/atoms/Container'
import Pagination from '@app/components/atoms/Pagination'
import ResultTable, {
  Props as ResultTableProps,
} from '@app/components/molecules/ResultTable'
import { FormValues as SearchFormValues } from '@app/components/organisms/ResultSearchForm'
import { GetUserResultsComponent } from '@app/queries'
import * as css from './style.scss'

const cx = classnames.bind(css)

export type Props = {
  screenName: string
  initialValues: SearchFormValues
  onSubmit: (values: SearchFormValues) => void
  defaultActivePage?: number
  numItemsPerPage?: number
  onPageChange?: ResultTableProps['onPageChange']
}

const ResultList: React.SFC<Props> = ({
  screenName,
  initialValues,
  onPageChange,
  numItemsPerPage = 20,
  defaultActivePage = 1,
}) => {
  const [formValues] = React.useState<SearchFormValues>(initialValues)
  const [activePage, setActivePage] = React.useState(defaultActivePage)

  const changePage = (newActivePage: number) => {
    setActivePage(newActivePage)

    if (onPageChange) {
      onPageChange(newActivePage)
    }
  }

  return (
    <Container>
      <div className={cx('result-list')}>
        <div className={cx('table-wrapper')}>
          <GetUserResultsComponent
            variables={{
              username: screenName,
              title: formValues.title,
              playStyle: formValues.playStyle,
              difficulties: formValues.difficulties,
              levels: formValues.levels,
            }}
          >
            {({ loading, error, data }) => {
              if (loading) {
                return 'loading'
              }
              if (error || !data || !data.searchMaps) {
                return <ErrorPage statusCode={404} />
              }

              const maps = _.map(data.searchMaps, ({ result, ...map }) => ({
                ...map,
                result,
              }))

              const totalPages = Math.ceil(maps.length / numItemsPerPage)
              const partialMaps = maps.slice(
                (activePage - 1) * numItemsPerPage,
                activePage * numItemsPerPage,
              )

              const pagination = (
                <Pagination
                  onPageChange={changePage}
                  totalPages={totalPages}
                  activePage={activePage}
                />
              )

              return (
                <>
                  <div className={cx('pagination', 'top')}>{pagination}</div>
                  <ResultTable
                    showBPI
                    maps={partialMaps}
                    screenName={screenName}
                  />
                  <div className={cx('pagination', 'bottom')}>{pagination}</div>
                </>
              )
            }}
          </GetUserResultsComponent>
        </div>
      </div>
    </Container>
  )
}

export default ResultList
