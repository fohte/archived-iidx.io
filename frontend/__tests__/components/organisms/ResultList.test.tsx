import * as React from 'react'
import { MockedProvider } from '@apollo/react-testing'
import * as renderer from 'react-test-renderer'

import { FormValues } from '@app/components/organisms/FilterForm'
import ResultList from '@app/components/organisms/ResultList'
import {
  GetUserResultsDocument,
  GetUserResultsQuery,
  PlayStyle,
} from '@app/queries'

const wait = (amount = 0) => new Promise(resolve => setTimeout(resolve, amount))

const screenName = 'username'

const initialValues: FormValues = {
  title: null,
  difficulties: [],
  levels: [],
}

const request = {
  query: GetUserResultsDocument,
  variables: {
    username: screenName,
    ...initialValues,
  },
}

describe('ResultList', () => {
  it('renders correctly', async () => {
    const data: GetUserResultsQuery = {
      searchMaps: { totalCount: 0, nodes: [] },
    }

    const mocks = [{ request, result: { data } }]

    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ResultList
          screenName={screenName}
          formValues={initialValues}
          playStyle={PlayStyle.Sp}
          activePage={1}
        />
      </MockedProvider>,
    )

    await renderer.act(async () => {
      // wait for response of the query
      await wait(0)
    })

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // FIXME: `An update to ResultList inside a test was not wrapped in act(...).`
  // という warning が出るが、どう修正するべきか分からないので一旦コメントアウトする
  // it('renders correctly when loading', () => {
  //   const data: GetUserResultsQuery = {
  //     searchMaps: { totalCount: 0, nodes: [] },
  //   }
  //
  //   const mocks = [{ request, result: { data } }]
  //
  //   const component = renderer.create(
  //     <MockedProvider mocks={mocks} addTypename={false}>
  //       <ResultList
  //         screenName={screenName}
  //         formValues={initialValues}
  //         playStyle={PlayStyle.Sp}
  //         activePage={1}
  //       />
  //     </MockedProvider>,
  //   )
  //
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })

  it('renders correctly if there are errors', async () => {
    const mocks = [{ request, error: new Error('something') }]

    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ResultList
          screenName={screenName}
          formValues={initialValues}
          playStyle={PlayStyle.Sp}
          activePage={1}
        />
      </MockedProvider>,
    )

    await renderer.act(async () => {
      // wait for response of the query
      await wait(0)
    })

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
