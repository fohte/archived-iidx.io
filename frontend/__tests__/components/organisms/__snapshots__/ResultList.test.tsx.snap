// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList renders correctly 1`] = `
<div
  className="result-list"
>
  <div
    className="form-wrapper"
  >
    <div
      className="box "
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-form"
        >
          <div
            className="button-group button-group"
          >
            <label>
              <input
                checked={true}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="SP"
              />
              <div
                className="button active "
                disabled={false}
              >
                SP
              </div>
            </label>
            <label>
              <input
                checked={false}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="DP"
              />
              <div
                className="button "
                disabled={false}
              >
                DP
              </div>
            </label>
          </div>
          <div
            className="search-input-group"
          >
            <input
              className="search-area"
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search..."
              type="text"
              value=""
            />
            <button
              className=""
              disabled={true}
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-search fa-w-16 "
                data-icon="search"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="table-wrapper"
  >
    <div
      className="pagination top"
    >
      <div
        className="pagination"
      >
        <a
          className="jump-to-first disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-double-left fa-w-14 "
            data-icon="angle-double-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="back disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-left fa-w-8 "
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <span
          className="page-info"
        >
          Page 
          <span
            className="active-page"
          >
            1
          </span>
           of
           
          0
        </span>
        <a
          className="forward disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-right fa-w-8 "
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="jump-to-last disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-double-right fa-w-14 "
            data-icon="angle-double-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div />
    <div
      className="pagination bottom"
    >
      <div
        className="pagination"
      >
        <a
          className="jump-to-first disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-double-left fa-w-14 "
            data-icon="angle-double-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="back disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-left fa-w-8 "
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <span
          className="page-info"
        >
          Page 
          <span
            className="active-page"
          >
            1
          </span>
           of
           
          0
        </span>
        <a
          className="forward disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-right fa-w-8 "
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="jump-to-last disabled"
          onClick={[Function]}
          rel="nofollow"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-double-right fa-w-14 "
            data-icon="angle-double-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResultList renders correctly if there are errors 1`] = `
<div
  className="result-list"
>
  <div
    className="form-wrapper"
  >
    <div
      className="box "
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-form"
        >
          <div
            className="button-group button-group"
          >
            <label>
              <input
                checked={true}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="SP"
              />
              <div
                className="button active "
                disabled={false}
              >
                SP
              </div>
            </label>
            <label>
              <input
                checked={false}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="DP"
              />
              <div
                className="button "
                disabled={false}
              >
                DP
              </div>
            </label>
          </div>
          <div
            className="search-input-group"
          >
            <input
              className="search-area"
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search..."
              type="text"
              value=""
            />
            <button
              className=""
              disabled={true}
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-search fa-w-16 "
                data-icon="search"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="table-wrapper"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#fff",
          "color": "#000",
          "display": "flex",
          "flexDirection": "column",
          "fontFamily": "-apple-system, BlinkMacSystemFont, Roboto, \\"Segoe UI\\", \\"Fira Sans\\", Avenir, \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
          "height": "100vh",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <div>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "body { margin: 0 }",
            }
          }
        />
        <h1
          style={
            Object {
              "borderRight": "1px solid rgba(0, 0, 0,.3)",
              "display": "inline-block",
              "fontSize": "24px",
              "fontWeight": 500,
              "margin": 0,
              "marginRight": "20px",
              "padding": "10px 23px 10px 0",
              "verticalAlign": "top",
            }
          }
        >
          404
        </h1>
        <div
          style={
            Object {
              "display": "inline-block",
              "height": "49px",
              "lineHeight": "49px",
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        >
          <h2
            style={
              Object {
                "fontSize": "14px",
                "fontWeight": "normal",
                "lineHeight": "inherit",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            This page could not be found
            .
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResultList renders correctly when loading 1`] = `
<div
  className="result-list"
>
  <div
    className="form-wrapper"
  >
    <div
      className="box "
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-form"
        >
          <div
            className="button-group button-group"
          >
            <label>
              <input
                checked={true}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="SP"
              />
              <div
                className="button active "
                disabled={false}
              >
                SP
              </div>
            </label>
            <label>
              <input
                checked={false}
                className="input"
                name="playStyle"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="DP"
              />
              <div
                className="button "
                disabled={false}
              >
                DP
              </div>
            </label>
          </div>
          <div
            className="search-input-group"
          >
            <input
              className="search-area"
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search..."
              type="text"
              value=""
            />
            <button
              className=""
              disabled={true}
              type="submit"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-search fa-w-16 "
                data-icon="search"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="table-wrapper"
  >
    loading
  </div>
</div>
`;
